Sources/Strand.swift

```diff
@@ -39,36 +39,32 @@ public class Strand {
 
     public init(closure: () -> Void) throws {
         let holder = Unmanaged.passRetained(StrandClosure(closure: closure))
-        let pointer = UnsafeMutablePointer<Void>(holder.toOpaque())
+        let pointer = UnsafeMutablePointer<Void>(OpaquePointer(bitPattern: holder))
+//        let pointer = UnsafeMutablePointer<Void>(holder.toOpaque())
 
-        #if swift(>=3.0)
-            #if os(Linux)
-                guard pthread_create(&pthread, nil, runner, pointer) == 0 else {
-                    holder.release()
-                    throw StrandError.threadCreationFailed
-                }
-            #else
-                guard pthread_create(&pthread, nil, runner, pointer) == 0 else {
-                    holder.release()
-                    throw StrandError.threadCreationFailed
-                }
-            #endif
-        #else
-            guard pthread_create(&pthread, nil, runner, pointer) == 0 else {
-                holder.release()
-                throw StrandError.threadCreationFailed
-            }
-        #endif
+        let runner: @convention(c) (UnsafeMutablePointer<Void>) -> UnsafeMutablePointer<Void>? = { arg in
```

Comment:
Not sure, the calls appeared to all be the same, just trying to get things building for now, then I can try it out. Is 2.2 something you're intent on supporting going forward?